presto-main/src/test/java/com/facebook/presto/type/TestDecimalOperators.java

```diff
@@ -195,25 +195,47 @@ public void testDivide()
         assertDecimalFunction("DECIMAL '200.00' / DECIMAL '0.3'", decimal("0666.67"));
         assertDecimalFunction("DECIMAL '200.00000' / DECIMAL '0.3'", decimal("0666.66667"));
         assertDecimalFunction("DECIMAL '200.00000' / DECIMAL '-0.3'", decimal("-0666.66667"));
-        assertDecimalFunction("DECIMAL '10' / DECIMAL '.00000001'", decimal("1000000000.00000000"));
-        assertDecimalFunction("DECIMAL '99999999999999999' / DECIMAL '1'", decimal("99999999999999999"));
-        assertDecimalFunction("DECIMAL '1' / DECIMAL '99999999999999999999999999999999999999'", decimal("0"));
-        assertDecimalFunction("DECIMAL '9' / DECIMAL '00000000000000003'", decimal("3"));
+        assertDecimalFunction("DECIMAL '999999999999999999' / DECIMAL '1'", decimal("999999999999999999"));
+        assertDecimalFunction("DECIMAL '9' / DECIMAL '000000000000000003'", decimal("3"));
         assertDecimalFunction("DECIMAL '9.0' / DECIMAL '3.0'", decimal("03.0"));
+        assertDecimalFunction("DECIMAL '999999999999999999' / DECIMAL '500000000000000000'", decimal("000000000000000002"));
+        assertDecimalFunction("DECIMAL '1' / DECIMAL '999999999999999999'", decimal("0"));
+        // round
+        assertDecimalFunction("DECIMAL '9' / DECIMAL '5'", decimal("2"));
```

Comment:
Okay, it may not be a problem because there's a syntax to allow you doing calculation on given precision:
`select cast(decimal '9' as decimal(3,2)) / cast(decimal '5' as decimal(3,2))` would return `1.80`.

I'd like to have tests covering operations on two decimals while both are of different precision/scale.